gorm.go

```diff
@@ -164,6 +164,80 @@ func Open(dialector Dialector, config *Config) (db *DB, err error) {
 	return
 }
 
+// DefaultConfig default config.
+func DefaultConfig() *Config {
+	c := &Config{
+		NamingStrategy: schema.NamingStrategy{},
+		Logger:         logger.Default,
+		NowFunc:        func() time.Time { return time.Now().Local() },
+		ClauseBuilders: map[string]clause.ClauseBuilder{},
+		Plugins:        map[string]Plugin{},
+		cacheStore:     &sync.Map{},
+	}
+
+	return c
+}
+
+// Apply apply option for Config.
+func (config *Config) Apply(opts ...ConfigOption) {
+	for _, o := range opts {
+		o(config)
+	}
+}
+
+// OpenWithOption create *db.DB with ConfigOption
+// return *db.DB,error.
+func OpenWithOption(dialector Dialector,
+	opts ...ConfigOption) (db *DB, err error) {
+	config := DefaultConfig()
+
+	// apply config option
+	config.Apply(opts...)
+
+	if dialector != nil {
+		config.Dialector = dialector
+	}
+
+	// create DB entry
```

Comment:
Below this seems to repeat the code from `Open()` ... why not just call it here instead of repeating?

Reply:
Here I really want to return a gorm DB similar to the Open method, because the default Open limits the parameters of Config, I want to customize the Config configuration.